code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -63,7 +63,7 @@ datum
 					if(current_list_element > reagent_list.len) current_list_element = 1
 					var/datum/reagent/current_reagent = reagent_list[current_list_element]
 
-					src.remove_reagent(current_reagent.id, 1)
+					src.remove_reagent(current_reagent.id, min(1, amount - total_transfered))

```

Comment:
> * Fixes the `remove_any` proc of reagents datums not properly removing fractional values.
  * This was causing lit smokables that you weren't smoking to burn off reagents 150% faster than intended.

The `smoke` proc uses `remove_any` to burn off reagents if a smokable isn't transferring them into a mob. Problem is, it tried to remove 0.4 per tick, but `remove_any` was removing 1 at a time.